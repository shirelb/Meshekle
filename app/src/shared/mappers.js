import strings from "./strings";

var loginScreenMapper = function (value) {
    switch (value) {
        case "userId doesn't exist!":
            return strings.loginScreenStrings.NO_SUCH_USER;
        default:
            return strings.loginScreenStrings.WRONG_CREDENTIALS;
    }
};

var mainScreenMapper = function (value) {
    switch (value) {
    }
};

var serviceProviderRolesMapper = function (value) {
    switch (value) {
        case "Admin":
            return "מנהלה";
        case "PhoneBookSecretary":
            return "מזכירות ספר טלפונים";
        case "ChoresSecretary":
            return "מזכירות תורנויות";
        case "AnnouncementsSecretary":
            return "מזכירות לוח מודעות";
        case "appointmentsHairDresser":
            return "מספרה";
        case "appointmentsDentist":
            return "מרפאת שיניים";
        default:
            return value;
    }
};

var serviceProviderRolesMapperHebToEng = function (value) {
    switch (value) {
        case "מנהלה":
            return "Admin";
        case "מזכירות ספר טלפונים":
            return "PhoneBookSecretary";
        case "מזכירות תורנויות":
            return "ChoresSecretary";
        case "מזכירות לוח מודעות":
            return "AnnouncementsSecretary";
        case "מספרה":
            return "appointmentsHairDresser";
        case "מרפאת שיניים":
            return "appointmentsDentist";
        default:
            return value;
    }
};

var appointmentRequestStatusMapper = function (value) {
    switch (value) {
        case "requested":
            return "ממתין לאישור";
        case "approved":
            return "אושר";
        case "rejected":
            return "נדחה";
        default:
            return value;
    }
};


var errorMapper = function (value) {
    switch (value.data.message) {
        case "userId doesn't exist!":
            return "משתמש לא קיים";
        case "user details not match, cant renew password":
            return "פרטי משתמש לא תואמים. לא ניתן לחדש סיסמא";
        case "AppointmentRequest not found!":
            return "בקשת תור לא קיימת";
        case "Appointment not found!":
            return "תור לא קיים";
        case "Incident not found!":
            return "תקלה לא קיימת";
        case "Authentication failed due to incorrect parameters":
            return "";
        case "Failed to authenticate token":
            return "";
        case "Token is valid!":
            return "";
        case "No token provided":
            return "";
        case "AppointmentRequest successfully added!":
            return "בקשת התור נוסםה בהצלחה";
        case "AppointmentsRequest successfully rejected!":
            return "בקשת התור נמחקה בהצלחה";
        case "Appointment successfully added!":
            return "התור נוסף בהצלחה";
        case "Appointment canceled successfully!":
            return "התור בוטל בהצלחה";
        case "Incident canceled successfully!":
            return "התקלה בוטלה בהצלחה";
        case "Incident already canceled !":
            return "התקלה כבר קיימת";
        case "Incident already resolved !":
            return "התקלה כבר נפתרה ונסגרה";
        case "Appointment already canceled !":
            return "התור כבר בוטל";
        case "Appointment already passed !":
            return "התור כבר עבר";
        case "Incident successfully added!":
            return "התקלה נוספה בהצלחה";
        case "User fields updated successfully!":
            return "פרטי המשתמש עודכנו בהצלחה";
        case "User not found!":
            return "משתמש לא קיים";
        case "Service provider already exists":
            return "נותן השירות כבר קיים";
        case "Service provider not found!":
            return "נותן השירות לא קיים";
        case "Service provider fields updated successfully!":
            return "פרטי נותן השירות עודכנו בהצלחה";
        case "Appointment way of type updated successfully!":
            return "דרך קביעת התורים של נותן השירות עודכנה בהצלחה";
        case "ServiceProvider successfully added!":
            return "נותן השירות נוסף בהצלחה";
        case "ServiceProvider role successfully added!":
            return "תחום אחיות חדש נוסף לנותן השירות בהצלחה";
        case "Role successfully deleted!":
            return "תחום האחריות נמחק בהצלחה";
        case "serviceProviders successfully deleted!":
            return "נותן השירות נמחק בהצלחה";
        case "User successfully added!":
            return "משתמש נוסך בהצלחה";
        case "User successfully deleted!":
            return "משתמש נמחק בהצלחה";
        case "Service Provider operationTime updated successfully!":
            return "זמן פעילות עודכן בהצלחה";
        case "Appointment status changed to cancelled successfully!":
            return "התור בוטל בהצלחה";
        case "Appointment updated successfully!":
            return "התור עודכן בהצלחה";
        case "Invalid input":
            return "קלט לא תקין";
        case "Invalid role input":
            return "תחום אחריות לא תקין";
        case "Invalid appointment way type input":
            return "דרך קביעת התורים לא תקינה";
        case "Invalid phone input":
            return "הטלפון לא תקין";
        case "Invalid email input":
            return "האימייל לא תקין";
        case "Invalid mailbox input":
            return "ת.ד. לא תקין";
        case "Invalid born date input":
            return "תאריך לידה לא תקין";
        case "User is already exists":
            return "המשתמש כבר קיים";
        case "Internal Server Error":
            return "קרתה שגיאה בשרת. אנא נסו שוב ואם השגיאה ממשיכה לקרות פנו לנציג המערכת";
        default:
            return "נסו שוב";
    }
};

export default {
    loginScreenMapper,
    mainScreenMapper,
    serviceProviderRolesMapper,
    appointmentRequestStatusMapper,
    serviceProviderRolesMapperHebToEng,
    errorMapper
};
