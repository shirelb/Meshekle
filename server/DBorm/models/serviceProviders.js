module.exports = (sequelize, type) => {
    return sequelize.define('ServiceProviders', {
        serviceProviderId:{
            type: type.INTEGER,
            primaryKey: true,
            autoIncrement: true,
            allowNull: false,
            required: true
        },
        role : {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        userId: {
            type: type.INTEGER,
            allowNull: false,
            required: true
        },
        operationTime: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        phoneNumber: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
    });
};