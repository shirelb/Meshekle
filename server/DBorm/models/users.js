module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};