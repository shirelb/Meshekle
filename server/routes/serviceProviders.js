var express = require('express');
var router = express.Router();
const Sequelize = require('sequelize');
const Op = Sequelize.Op;
const {Users, ServiceProviders, AppointmentRequests, Appointments, Incidents, Events} = require('../DBorm/DBorm');

/* GET users listing. */
router.get('/', function(req, res, next) {
  res.send('respond with a resource serviceProviders');
});


/* POST user . */
router.post('/users/add', function (req, res, next) {
    Users.create({
        userId: req.body.userId,
        fullname: req.body.fullname,
        password: req.body.password,
        email: req.body.email,
        mailbox: req.body.mailbox,
        cellphone: req.body.cellphone,
        phone: req.body.phone,
        bornDate: new Date(req.body.bornDate),
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* POST user . */
router.post('/add', function (req, res, next) {
    ServiceProviders.create({
        serviceProviderId: req.body.serviceProviderId,
        userId: req.body.userId,
        role: req.body.role,
        operationTime: req.body.operationTime,
        phoneNumber: req.body.phoneNumber,
        appointmentWayType: req.body.appointmentWayType,
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

module.exports = router;
