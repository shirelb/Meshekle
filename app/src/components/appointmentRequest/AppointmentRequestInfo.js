import React, {Component} from 'react';
import {Modal, ScrollView, StyleSheet, View} from "react-native";
import {FormLabel, Text} from "react-native-elements";
import Button from "../submitButton/Button";
import {List} from "react-native-paper";
import mappers from "../../shared/mappers";
import colors from "../../shared/colors";


export default class AppointmentRequestInfo extends Component {
    constructor(props) {
        super(props);

        this.state = {
            modalVisible: this.props.modalVisible,
            optionalTimes: JSON.parse(this.props.appointmentRequest.optionalTimes),
        };

    }

    componentWillReceiveProps(nextProps, nextContext) {
        this.setState({
            modalVisible: this.props.modalVisible,
        })
    }

    setModalVisible(visible) {
        this.setState({modalVisible: visible,appointmentRequestDetails: {},});
    }

    render() {
        const appointmentRequest = this.props.appointmentRequest;
        console.log('appointmentRequest ', appointmentRequest);

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={this.state.modalVisible}
                onRequestClose={() => {
                    this.setModalVisible(!this.state.modalVisible);
                    this.props.closeAppointmentRequestInfo();
                }}
            >
                <View style={{marginTop: 20}}>
                    <ScrollView>
                        <Text h4 style={styles.textTitle}>בקשת תור</Text>
                        {/*<Header
                                backgroundColor={'white'}
                                // leftComponent={{ icon: 'menu', color: '#fff' }}
                                centerComponent={{ text: 'בקשת תור', style: { color: '#050505',fontWeight: 'bold',fontSize:26 } }}
                                // rightComponent={{ icon: 'home', color: '#fff' }}
                            />*/}

                        <FormLabel>נותן שירות</FormLabel>
                        <Text style={{marginLeft: 10}}>
                            {appointmentRequest.serviceProviderFullname}
                        </Text>

                        <FormLabel> ענף</FormLabel>
                        <Text style={{marginLeft: 10}}>
                            {mappers.serviceProviderRolesMapper(appointmentRequest.AppointmentDetail.role)}
                        </Text>

                        <FormLabel> סטאטוס</FormLabel>
                        <Text style={{marginLeft: 10}}>
                            {mappers.appointmentRequestStatusMapper(appointmentRequest.status)}
                        </Text>

                        <FormLabel>נושא</FormLabel>
                        <Text style={{marginLeft: 10}}>
                            {JSON.parse(appointmentRequest.AppointmentDetail.subject).join(", ")}
                        </Text>

                        <FormLabel> תאריכים ושעות אופציונאליים</FormLabel>
                        <List.Section>
                            {Array.isArray(this.state.optionalTimes) &&
                            this.state.optionalTimes.map((item, index) => {
                                return <List.Accordion
                                    key={index}
                                    // title={moment(item.startDateAndTime).format('HH:mm') + '-' + moment(item.endDateAndTime).format('HH:mm')}
                                    title={item.date}
                                    // description={item.AppointmentDetail.role + ',' + item.AppointmentDetail.serviceProviderId}
                                    // left={props => <List.Icon {...props} icon="perm-contact-calendar"/>}
                                    // expanded={this.state.expanded[item.date]}
                                    expanded={item.expanded}
                                    onPress={() => {
                                        // let expanded = this.state.expanded;
                                        // expanded[item.date] = !expanded[item.date];
                                        // this.setState({expanded: expanded})
                                        let datesAndTimes = this.state.optionalTimes;
                                        datesAndTimes[index].expanded = !datesAndTimes[index].expanded;
                                        this.setState({optionalTimes: datesAndTimes})
                                    }}
                                >
                                    {Array.isArray(item['hours']) && item['hours'] ?
                                        item['hours'].map((hour, j) => {
                                            return <List.Item
                                                key={j}
                                                title={hour.startHour + '-' + hour.endHour}
                                                // description={item.remarks}
                                                containerStyle={{borderBottomWidth: 0}}
                                                onPress={() => console.log("item was presssed!!  ", item)}
                                            />
                                        }) : null
                                    }
                                </List.Accordion>
                            })
                            }
                        </List.Section>

                        <FormLabel>הערות</FormLabel>
                        <Text style={{marginLeft: 10}}>
                            {appointmentRequest.notes === "" ? "אין הערות" : appointmentRequest.notes}
                        </Text>

                        <View style={{marginTop: 20}}>
                            <Button
                                label='מחק'
                                onPress={() => {
                                    this.setModalVisible(!this.state.modalVisible);
                                    this.props.cancelAppointmentRequest(appointmentRequest);
                                    this.props.closeAppointmentRequestInfo();
                                }}
                                color={colors.TORCH_RED}
                            />
                            <Button
                                label='חזור'
                                onPress={() => {
                                    this.setModalVisible(!this.state.modalVisible);
                                    this.props.closeAppointmentRequestInfo();
                                }}
                            />
                        </View>

                    </ScrollView>
                </View>
            </Modal>
        );
    }
}

const styles = StyleSheet.create({
    textTitle: {
        textAlign: 'center',
        color: '#050505',
        fontWeight: 'bold',
    }
});
