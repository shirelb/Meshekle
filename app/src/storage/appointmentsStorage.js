import axios from "axios";
import {APP_SOCKET, SERVER_URL} from "../shared/constants";


var getUserAppointmentRequests = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'requested'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointment requests error ', error)
        });
};

var getUserAppointments = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'set'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments error ', error)
        });
};

var getUserAppointmentById = function (userId, userHeaders, eventId) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {status: 'set', appointmentId: eventId}
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments by id error ', error)
        });
};

var postUserAppointmentRequest = function (userId, serviceProvider, appointmentRequest, userHeaders) {
    return axios.post(`${SERVER_URL}/api/appointmentRequests/user/request`,
        {
            userId: userId,
            serviceProviderId: serviceProvider.serviceProviderId,
            role: serviceProvider.role,
            availableTime: appointmentRequest.availableTime,
            notes: appointmentRequest.notes ? appointmentRequest.notes : '',
            subject: JSON.stringify(appointmentRequest.subject)
        },
        {
            headers: userHeaders,
        })
        .then(response => {
            APP_SOCKET.emit('userPostAppointmentRequests', {
                serviceProviderId: serviceProvider.serviceProviderId
            });

            return response;
        })
        .catch(error => {
            console.log('post user appointment request error ', error)
        });
};

var cancelAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/user/reject`,
        {
            userId: appointmentRequest.AppointmentDetail.clientId,
            appointmentRequestId: appointmentRequest.requestId,
        },
        {
            headers: headers,
            params: {
                status: 'rejected'
            },
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointmentRequests', {
                serviceProviderId: appointmentRequest.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('reject appointment request error ', error);
        });
};

var cancelAppointmentById = (appointment, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/user/cancel/userId/${appointment.AppointmentDetail.clientId}/appointmentId/${appointment.appointmentId}`,
        {},
        {
            headers: headers
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointment', {
                serviceProviderId: appointment.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('cancel appointment error ', error);
        });
};


export default {
    getUserAppointments,
    getUserAppointmentById,
    postUserAppointmentRequest,
    getUserAppointmentRequests,
    cancelAppointmentRequestById,
    cancelAppointmentById,
};
