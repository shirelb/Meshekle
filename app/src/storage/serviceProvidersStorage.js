import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
