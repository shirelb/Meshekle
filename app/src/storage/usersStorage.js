import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
