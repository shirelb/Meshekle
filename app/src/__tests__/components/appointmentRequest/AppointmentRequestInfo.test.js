import React from 'react';

import {Modal} from "react-native";
import {CheckBox, FormInput, FormLabel, FormValidationMessage, Text} from "react-native-elements";
import {List} from "react-native-paper";
import {shallow} from "enzyme/build";
import AppointmentRequestInfo from "../../../components/appointmentRequest/AppointmentRequestInfo";
import usersStorage from "../../../storage/usersStorage";
import phoneStorage from "react-native-simple-store";
import users from "../../jsons/users";
import serviceProvidersStorage from "../../../storage/serviceProvidersStorage";
import serviceProviders from "../../jsons/serviceProviders";
import appointmentsStorage from "../../../storage/appointmentsStorage";
import user013637605AppointmentRequests from "../../jsons/user013637605AppointmentRequests";
import mappers from "../../../shared/mappers";

jest.mock("../../../storage/serviceProvidersStorage");
jest.mock("../../../storage/usersStorage");
jest.mock("react-native-simple-store");


describe("AppointmentRequestInfo should", () => {
    let wrapper = null;
    let componentInstance = null;
    const userTest = users[2];
    const appointmentRequestTest = user013637605AppointmentRequests[0];
    const userIdTest = "972350803";
    const userFullnameTest = "Dion Revance";
    const props = {
        modalVisible: true,
        appointmentRequest: appointmentRequestTest,
        cancelAppointmentRequest: jest.fn(),
        closeAppointmentRequestInfo: jest.fn(),
    };
    const mockStore = {
        userData: {
            serviceProviderId: "549963652",
            userId: "549963652",
            token: "some token"
        }
    };
    const navigation = {navigate: jest.fn(), dispatch: jest.fn()};

    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    phoneStorage.update = jest.fn().mockImplementation((key, value) => Promise.resolve(mockStore[key] = value));
    usersStorage.getUsers = jest.fn().mockImplementation(() => Promise.resolve(users));
    usersStorage.getUserByUserID = jest.fn().mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviders = jest.fn().mockResolvedValue(serviceProviders);
    serviceProvidersStorage.getServiceProviderUserDetails = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve({data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));
    appointmentsStorage.getUserAppointmentRequests = jest.fn().mockImplementation(() => Promise.resolve(user013637605AppointmentRequests));
    appointmentsStorage.postUserAppointmentRequest = jest.fn().mockImplementation(() => Promise.resolve());

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('match snapshot', async () => {
        wrapper = shallow(<AppointmentRequestInfo {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("render what the user see", async () => {
        wrapper = await shallow(<AppointmentRequestInfo {...props} />);
        componentInstance = wrapper.instance();

        await wrapper.update();

        expect(wrapper.find(Modal)).toHaveLength(1);
        expect(wrapper.find(Text)).toHaveLength(6);
        expect(wrapper.find(Text).at(0).props().children).toEqual('בקשת תור');
        expect(wrapper.find(Text).at(1).props().children).toEqual(appointmentRequestTest.serviceProviderFullname);
        expect(wrapper.find(Text).at(2).props().children).toEqual(mappers.serviceProviderRolesMapper(appointmentRequestTest.AppointmentDetail.role));
        expect(wrapper.find(Text).at(3).props().children).toEqual(mappers.appointmentRequestStatusMapper(appointmentRequestTest.status));
        expect(wrapper.find(Text).at(4).props().children).toEqual(JSON.parse(appointmentRequestTest.AppointmentDetail.subject).join(", "));
        expect(wrapper.find(Text).at(5).props().children).toEqual(appointmentRequestTest.notes);
        expect(wrapper.find('FormLabel')).toHaveLength(6);
        expect(wrapper.find('FormLabel').at(0).props().children).toEqual('נותן שירות');
        expect(wrapper.find('FormLabel').at(1).props().children).toEqual(" ענף");
        expect(wrapper.find('FormLabel').at(2).props().children).toEqual(" סטאטוס");
        expect(wrapper.find('FormLabel').at(3).props().children).toEqual("נושא");
        expect(wrapper.find('FormLabel').at(4).props().children).toEqual(' תאריכים ושעות אופציונאליים');
        expect(wrapper.find('FormLabel').at(5).props().children).toEqual('הערות');
        expect(wrapper.find(List.Section)).toHaveLength(1);
        expect(wrapper.find(List.Accordion)).toHaveLength(2);
        expect(wrapper.find(List.Item)).toHaveLength(4);
        expect(wrapper.find('Button')).toHaveLength(2);
        expect(wrapper.find('Button').at(0).props().label).toEqual('מחק');
        expect(wrapper.find('Button').at(1).props().label).toEqual('חזור');
    });

    it("mounted with the right data", async () => {
        expect(componentInstance.state.modalVisible).toEqual(true);
        expect(componentInstance.props.appointmentRequest).toEqual(appointmentRequestTest);
    });

    it("handle setModalVisible", async () => {
        wrapper = shallow(<AppointmentRequestInfo {...props} />);
        componentInstance = wrapper.instance();

        expect(componentInstance.state.modalVisible).toEqual(true);

        await componentInstance.setModalVisible(false);
        expect(componentInstance.state.modalVisible).toEqual(false);
    });

});