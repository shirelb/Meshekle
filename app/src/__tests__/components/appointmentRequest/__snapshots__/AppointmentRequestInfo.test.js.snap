// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentRequestInfo should match snapshot 1`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <ScrollViewMock>
      <TextElement
        h4={true}
        style={
          Object {
            "color": "#050505",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        בקשת תור
      </TextElement>
      <FormLabel>
        נותן שירות
      </FormLabel>
      <TextElement
        style={
          Object {
            "marginLeft": 10,
          }
        }
      />
      <FormLabel>
         ענף
      </FormLabel>
      <TextElement
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        מספרה
      </TextElement>
      <FormLabel>
         סטאטוס
      </FormLabel>
      <TextElement
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        נדחה
      </TextElement>
      <FormLabel>
        נושא
      </FormLabel>
      <TextElement
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        פן, תספורת, צבע, גוונים
      </TextElement>
      <FormLabel>
         תאריכים ושעות אופציונאליים
      </FormLabel>
      <withTheme(List.Section)>
        <withTheme(List.Accordion)
          key="0"
          onPress={[Function]}
          title="2019-03-29"
        >
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            key="0"
            onPress={[Function]}
            title="4:51-20:55"
          />
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            key="1"
            onPress={[Function]}
            title="7:24-23:41"
          />
        </withTheme(List.Accordion)>
        <withTheme(List.Accordion)
          key="1"
          onPress={[Function]}
          title="2018-06-28"
        >
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            key="0"
            onPress={[Function]}
            title="8:45-22:50"
          />
          <withTheme(List.Item)
            containerStyle={
              Object {
                "borderBottomWidth": 0,
              }
            }
            key="1"
            onPress={[Function]}
            title="1:21-17:10"
          />
        </withTheme(List.Accordion)>
      </withTheme(List.Section)>
      <FormLabel>
        הערות
      </FormLabel>
      <TextElement
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        Nulla tempus.
      </TextElement>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <Button
          color="#F8262F"
          label="מחק"
          onPress={[Function]}
        />
        <Button
          label="חזור"
          onPress={[Function]}
        />
      </View>
    </ScrollViewMock>
  </View>
</Component>
`;
